### PR TITLE
[SYCL][ESIMD] Fix GenXIntrinsic lowering for vectors used as scalars

### PR DESCRIPTION
**Pull request by sarnex**
In most cases, we see the following IR for vector spirv intrinsics:
```
%0 = load <3 x i64>, <3 x i64> addrspace(4)* @__spirv_BuiltInGlobalInvocationId, align 32
%1 = extractelement <3 x i64> %0, i64 0
// some use of %1, ex:
%2 = icmp ult i64 %0, 50
```

However, if only the first element is used, this may be optimized to:
```
%0 = load i64, addrspace(4)* @__spirv_BuiltInGlobalInvocationId, align 32
// some use of %0, ex:
%1 = icmp ult i64 %0, 50
```

In the latter case, if we try to insert a cast due to different types between the real intrinsic type and the global variable type, make sure to use the load type, because the use type is arbitrary code and may not be the same as the global variable type.

This is fine for the case with ExtractElement because in that case the type of the EE will be based on the global variable type.

### CODE REVIEW

**Comment by sarnex** [comment]
```
Failed Tests (1):
  SYCL :: Plugin/level_zero_imm_cmdlist_per_thread.cpp
```

```

********************
Failed Tests (1):
  SYCL :: Regression/same_unnamed_kernels.cpp
```

not related

File: llvm/lib/SYCLLowerIR/ESIMD/LowerESIMD.cpp [hunk 1]
```diff
@@ -1086,7 +1087,8 @@ static uint64_t getIndexFromExtract(ExtractElementInst *EEI) {
 /// of vector load. The parameter \p IsVectorCall tells what version of GenX
 /// intrinsic (scalar or vector) to use to lower the load from SPIRV global.
 static Instruction *generateGenXCall(Instruction *EEI, StringRef IntrinName,
-                                     bool IsVectorCall, uint64_t IndexValue) {
+                                     bool IsVectorCall, uint64_t IndexValue,
+                                     Type *UseType = nullptr) {
```
Comment (v-klochkov) [inline llvm/lib/SYCLLowerIR/ESIMD/LowerESIMD.cpp:1091]: nit: with the new changes 'UsedType' is always passed to this function.
```suggestion
                                     Type *UseType) {
```

**Review by v-klochkov — APPROVED** [review]

File: llvm/lib/SYCLLowerIR/ESIMD/LowerESIMD.cpp (hunk 1 above)
Comment (sarnex) [reply llvm/lib/SYCLLowerIR/ESIMD/LowerESIMD.cpp:1091]: good catch, thx

**Comment by v-klochkov** [comment]
This fail is unrelated to this PR with changes in ESIMD lowering.
  SYCL :: Regression/same_unnamed_kernels.cpp

**Pull request merged**